server/auth/index.ts

```diff
@@ -38,31 +41,76 @@ export async function requireUserAuth(req: AuthRequest, res: Response, next: Nex
   });
 
   if (!user) {
-    throw new ForbiddenError(`User '${id}' is not registered.`, true);
+    user = new User();
+    user.edipi = id;
+  }
+
+  if (user.root_admin) {
+    user.roles = (await Org.find()).map(org => Role.admin(org));
   }
 
   req.appUser = user;
 
   next();
 }
 
+export async function requireRegisteredUser(req: ApiRequest, res: Response, next: NextFunction) {
+  if (req.appUser.is_registered) {
+    return next();
+  }
+  throw new ForbiddenError('User is not registered.');
+}
+
 export async function requireRootAdmin(req: ApiRequest, res: Response, next: NextFunction) {
   if (req.appUser.root_admin) {
     return next();
   }
   throw new ForbiddenError('User does not have sufficient privileges to perform this action.');
 }
 
-export function requireRolePermission(action: (role: Role) => boolean) {
-  return async (req: ApiRequest<OrgParam>, res: Response, next: NextFunction) => {
-    const org = parseInt(req.params.orgId);
-    const user = req.appUser;
-    if (org && user) {
-      const orgRole = user.roles.find(role => role.org.id === org);
-      if (user.root_admin || (orgRole && action(orgRole))) {
-        return next();
+export async function requireOrgAccess(req: any, res: Response, next: NextFunction) {
+  let orgId = -1;
+  if (req.params.orgId) {
+    orgId = parseInt(req.params.orgId);
+  } else if (req.query.orgId) {
+    orgId = parseInt(req.query.orgId);
+  } else if (req.cookies.hasOwnProperty('orgId')) {
+    orgId = parseInt(req.cookies.orgId);
+  }
+  if (orgId === -1) {
```

Comment:
Right now it looks like someone could pass in a negative number or some text for `orgId` and it would make it past the error check...

I think think this might be a little tighter if `orgId` is initially undefined or null and we switch the first check to `if (orgId == null)`, then we add another check `if (isNaN(orgId) || orgId < 0)` and throw a different "Invalid org id" error in that case.